crates/tx5/src/back_buf/imp/imp_webrtc_rs.rs

```diff
@@ -0,0 +1,115 @@
+use crate::*;
+use bytes::{Bytes, BytesMut, BufMut, Buf};
+
+pub struct ImpWriter {
+    buf: BytesMut,
+}
+
+impl ImpWriter {
+    #[inline]
+    pub fn new() -> Result<Self> {
+        Ok(Self {
+            buf: BytesMut::new(),
+        })
+    }
+
+    #[inline]
+    pub fn finish(self) -> Imp {
+        Imp { buf: self.buf }
+    }
+}
+
+impl std::io::Write for ImpWriter {
+    #[inline]
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let mut writer = self.buf.writer();
+        writer.write(buf)
+    }
+
+    #[inline]
+    fn write_vectored(
+        &mut self,
+        bufs: &[std::io::IoSlice<'_>],
+    ) -> std::io::Result<usize> {
+        let mut writer = self.buf.writer();
+        writer.write_vectored(bufs)
+    }
+
+    #[inline]
+    fn write_all(&mut self, buf: &[u8]) -> std::io::Result<()> {
+        let mut writer = self.buf.writer();
+        writer.write_all(buf)
+    }
+
+    #[inline]
+    fn flush(&mut self) -> std::io::Result<()> {
+        let mut writer = self.buf.writer();
+        writer.flush()
+    }
+}
+
+pub struct Imp {
+    pub(crate) buf: BytesMut,
```

Comment:
Would it work to have this one be a non-mut `Bytes`, and freeze in the `from_*` constructors? I don't *think* we ever mutate the backbufs and then the try_clones would be cheaper.

Reply:
yep, appears to work

Reply:
have I resolved?